core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/MetadataSetupService.java

```diff
@@ -361,6 +367,24 @@ public void saveLoadBalancerMetadata(Stack stack, Iterable<CloudLoadBalancerMeta
                 loadBalancerEntry.setIp(cloudLoadBalancerMetadata.getIp());
                 loadBalancerEntry.setType(cloudLoadBalancerMetadata.getType());
                 String endpoint = loadBalancerConfigService.generateLoadBalancerEndpoint(stack);
+
+                List<StackStatusView> stoppedDatalakes = stackService.getByEnvironmentCrnAndStackType(stack.getEnvironmentCrn(), StackType.DATALAKE)
+                        .stream().filter(s -> s.getStatus().getStatus().isStopState()).collect(Collectors.toList());
+                if (!stoppedDatalakes.isEmpty()) {
+                    LOGGER.info("Using old datalake endpoint name for resized datalake: {}, env: {}", stack.getName(), stack.getEnvironmentCrn());
+                    if (stoppedDatalakes.size() > 1) {
+                        String ids = stoppedDatalakes.stream().map(StackStatusView::getId).map(Object::toString).collect(Collectors.joining(","));
+                        LOGGER.warn("more than one datalake found to resize from: {}", ids);
+                    }
+                    Long oldId = stoppedDatalakes.get(0).getId();
+
+                    Set<LoadBalancer> oldLoadbalancers = loadBalancerPersistenceService.findByStackId(oldId);
+                    if (oldLoadbalancers.isEmpty()) {
+                        Stack oldStack = stackService.getByIdWithGatewayInTransaction(oldId);
+                        endpoint = oldStack.getPrimaryGatewayInstance().getShortHostname();
+                    }
+                }
+
```

Comment:
This logic makes a wild guess. This logic could do go wrong if there is another stack in stop state.
Please add a filter on the name to see if the name is of format stackname-<timestamp>. 
Also, add comments on why this new behavior is introduced and the use it impacts.

Below is a sample code for reference. Make changes as needed.


    private String getLoadBalancerEndpoint(Stack stack) {
        List<StackStatusView> stoppedDatalakes = stackService.getByEnvironmentCrnAndStackType(stack.getEnvironmentCrn(), StackType.DATALAKE)
                .stream().filter(s -> s.getStatus().getStatus().isStopState())
                .filter(s -> {
                    int delimiterPos = s.getName().lastIndexOf("_");
                    if (delimiterPos < 0) {
                        return false;
                    }
                    if(s.getName().substring(0, delimiterPos).equals(stack.getName())) {
                        return true;
                    }
                    return  false;
                })
                .collect(Collectors.toList());
        if(stoppedDatalakes.isEmpty()) {
            return loadBalancerConfigService.generateLoadBalancerEndpoint(stack);
        } else {
            Long oldId = stoppedDatalakes.get(0).getId();
            Set<LoadBalancer> oldLoadbalancers = loadBalancerPersistenceService.findByStackId(oldId);
            if (oldLoadbalancers.isEmpty()) {
                Stack oldStack = stackService.getByIdWithGatewayInTransaction(oldId);
                return oldStack.getPrimaryGatewayInstance().getShortHostname();
            } else {
                return loadBalancerConfigService.generateLoadBalancerEndpoint(stack);
            }
        }
    }